cucapra/filament
# Use docker images in testing

Tries using dockerfile for tests. ~~Currently uses `latest` because we don't have tagged versions.~~

This splits the `test` action into `3` steps:
1. Use `git log` to get the hash of the last commit that edited `Dockerfile`, and also check if that ghcr tag already exists on the remote.
2. Conditionally run a build-push action if it doesn't exist, which just builds and pushes the docker image of that specific commit with the right tag.
3. Run test using that docker image.
